### PR TITLE
RTC alarm & calibration output

### PR DESCRIPTION
**Pull request by dotcypress**
This PR contains:

- RTC alarm support
- Calibration output pins support
-  LSE bug fix

### CODE REVIEW

File: src/rtc.rs [hunk 1]
```diff
@@ -111,6 +190,178 @@ impl Rtc {
...
+                    .hu()
+                    .bits(hu)
+                    .mnt()
+                    .bits(mt)
+                    .mnu()
+                    .bits(mu)
+                    .st()
+                    .bits(st)
+                    .su()
+                    .bits(su)
```
Comment (burrbull) [inline src/rtc.rs:279]:
```suggestion
                w.wdsel().bit(alarm.use_weekday);
                w.msk1().bit(alarm.seconds.is_none());
                w.msk2().bit(alarm.minutes.is_none());
                w.msk3().bit(alarm.hours.is_none());
                w.msk4().bit(alarm.day.is_none());
                w.dt().bits(dt);
                w.du().bits(du);
                w.ht().bits(ht);
                w.hu().bits(hu);
                w.mnt().bits(mt);
                w.mnu().bits(mu);
                w.st().bits(st);
                w.su().bits(su)
```

Comment (burrbull) [reply src/rtc.rs:279]: same for other big blocks

File: src/rtc.rs [hunk 2]
```diff
@@ -111,6 +190,178 @@ impl Rtc {
     pub fn get_week_day(&self) -> u8 {
         self.rb.dr.read().wdu().bits()
     }
+
+    pub fn set_alarm_a(&mut self, alarm: Alarm) {
```
Comment (burrbull) [inline src/rtc.rs:194]:
```suggestion
    pub fn set_alarm_a(&mut self, alarm: impl Into<Alarm>) {
        let alarm = alarm.into();
```

File: src/rtc.rs [hunk 3]
```diff
@@ -1,42 +1,121 @@
...
 use crate::time::*;
 
+#[derive(Debug, PartialEq, Eq)]
+pub enum RtcHourFormat {
+    H24,
+    H12,
+}
+
+#[derive(Debug, PartialEq, Eq)]
+pub enum RtcCalibrationFreqency {
```
Comment (burrbull) [inline src/rtc.rs:14]: typo

**Comment by burrbull** [comment]
Do you plan to support https://docs.rs/time/0.3.17/time/struct.Time.html?

**Comment by dotcypress** [comment]
@burrbull yes, now I have plans for `time` crate, but probably this is will be implemented in separate PR.

**Review by burrbull — APPROVED** [review]

**Pull request merged**